### ISSUE TITLE
dartdoc shows the wrong parameters for new-style generic function types

### ISSUE THREAD

**Issue by jcollins-g** [dart-lang/dartdoc#1472]
A function that looks like this:
```
/// A typedef with the new style generic function syntax.
typedef NewGenericTypedef<T> = List<S> Function<S>(T, int, bool);
```
is currently documented as having parameters (T, T, T).

This is caused by analyzer producing ParameterElementImpls that compare equally to each other, subverting dartdoc's internal object cache.   dart-lang/sdk#30146

**Comment by jcollins-g** [comment]
This also occurs for types created as functions with parameters defined outside of typedefs, for example:

```
  void Function(T1, T2) bindBinaryCallbackGuarded<T1, T2>(
      void callback(T1 argument1, T2 argument2));
```

**Comment by jcollins-g** [comment]
Fixed in #1524.
